File: models/UserRole.js
const {DataTypes} = require('sequelize');
const {sequelize} = require('../config/database');
const AuthUser = require('./AuthUser');
const Role = require('./Role');

const UserRole = sequelize.define('UserRole', {
    user_id: {
        type: DataTypes.BIGINT,          // removed .UNSIGNED
        primaryKey: true,
        allowNull: false,
        references: {
            model: AuthUser,               // model reference (not string)
            key: 'user_id'
        },
        onDelete: 'CASCADE'
    },
    role_id: {
        type: DataTypes.INTEGER,         // removed .UNSIGNED
        primaryKey: true,
        allowNull: false,
        references: {
            model: Role,
            key: 'role_id'
        },
        onDelete: 'CASCADE'
    }
}, {
    tableName: 'user_roles',
    schema: 'dbo',
    timestamps: false
});

module.exports = UserRole;

File: models/user_role.py
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy import Integer, BigInteger, ForeignKey, Table, MetaData
from .base import Base


class UserRole(Base):
    __tablename__ = "user_roles"
    __table_args__ = {"schema": "dbo"}

    user_id: Mapped[int] = mapped_column(
        BigInteger,
        ForeignKey("dbo.users.user_id", ondelete="CASCADE"),
        primary_key=True,
    )
    role_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("dbo.roles.role_id", ondelete="CASCADE"), primary_key=True
    )
